language/vm/src/check_bounds.rs

```diff
@@ -182,38 +183,56 @@ impl<'a> BoundsChecker<'a> {
         Ok(())
     }
 
-    fn check_function_def(&self, function_def: &FunctionDefinition) -> VMResult<()> {
+    fn check_function_def(
+        &self,
+        function_def_idx: usize,
+        function_def: &FunctionDefinition,
+    ) -> VMResult<()> {
         check_bounds_impl(&self.module.function_handles, function_def.function)?;
         for ty in &function_def.acquires_global_resources {
             check_bounds_impl(&self.module.struct_defs, *ty)?;
         }
-        self.check_code(function_def)
+        self.check_code(function_def_idx, function_def)
     }
 
-    fn check_code(&self, function_def: &FunctionDefinition) -> VMResult<()> {
+    fn check_code(
+        &self,
+        function_def_idx: usize,
+        function_def: &FunctionDefinition,
+    ) -> VMResult<()> {
         let code_unit = match &function_def.code {
             Some(code) => code,
             None => return Ok(()),
         };
         check_bounds_impl(&self.module.signatures, code_unit.locals)?;
 
-        let type_param_count = match self
-            .module
-            .function_handles
-            .get(function_def.function.into_index())
-        {
-            Some(fh) => fh.type_parameters.len(),
-            None => 0,
-        };
-        // if there are locals check that the type parameters in local signature are in bounds.
+        debug_assert!(function_def.function.into_index() < self.module.function_handles.len());
```

Comment:
not clear about the 2 `debug_assert`. Where those conditions we had checked before?

Reply:
Yes, they should have been checked before.